### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by nzjrs**
Current understanding of the whys and the hows

### CODE REVIEW

**Comment by nzjrs** [comment]
@sdvillal thoughts?

**Comment by nzjrs** [comment]
I'd like to put the knowledge about WTF is going on with blas features and build numbers >200 if possible.

**Comment by sdvillal** [comment]
I think this is too emotional ;-)

Regarding the 200 in the opencv build number, I do not think it is something we need to write about here. As you correctly write, if conda do what we ask it to do, build numbers in other channels do not matter.

The reason they use strange build numbers in the 100s is simply a matter of selecting a default build when using the deprecated (but still in place) conda-features way to specify package variants. So if you have two blas variants, lets say mkl and openblas, and you want openblas to be the default in your stack, you can disambiguate any possible ambiguity using the build number: 0...200 for mkl, 200...whatever for openblas. This spreads to downstream dependencies, so if you want that when you do ```conda install numpy opencv``` the default variant is the openblas one, you also give these numpy and opencv variants a number in the 200. That is why conda-forge opencv builds, which by the way only have one variant, inherit the 200 number, which in turn makes them gain precedence over other opencv packages with the same version but lower build number - like ours. I'm actually surprised to see that conda-forge numpy has recently stopped using the 200 hack and simply put the variant name in their build string (but still require the damn blas_openblas feature, which I understand is a deprecated mechanism and not anymore in use in defaults).

This is because of conda missing a good mechanism to declare package variants - and more genral predicates on packages relationships, like "provides" and "conflicts". Right now several packages (for example, tensorflow) have moved to a way of specifying variants that is better than features while still not requiring changes in conda. The method is quite complex, so if you want I could explain at another moment, but it is enough to say that, while much better, it is still far from perfect.

In our packages we use the build string, which is simple and works like a charm at the moment. Truth to be told, it has its drawbacks (you essenitally need to regexp and there is no way to specify a preferred build unless we would use build number or prefixes hacks), but beyond that, it is a sane choice for simplicity and that guarantees no cruft is left when uninstalling / replacing packages.

File: README.md [hunk 1]
```diff
@@ -18,3 +18,41 @@ error instead of silently succeeding (which `conda env` would do, thus
...
+
+If you want to use conda envs `environment.yaml` then the only way to guarentee that you get the build from the channel you requested is to use hashed build strings. But if you are doing this, you might as well use explicit environment / lists of packages anyway.
+
+A 'known' [workaround](https://github.com/conda/conda/issues/6385#issuecomment-361787360) has been to just call `conda create` **not conda env** with all the packages in the environment yaml. **That's what this tool does**
+
+TL/DR: let's use the (hashed) build string to ensure conda gets our packages until channel pins in environment.yaml files get respected. From there, we should also try to use "locked" (aka "explicit") environments - although that is more of an upcoming conda feature than a reality in the conda 4.5 series.
+
+In principle this is fixed in conda 4.6, but moving to a major new version of a tool which you already do not have the most trust in, is a scary affair. For now this tool does the same thing.
+
+## Other Notes
```
Comment (sdvillal) [inline README.md]: I think this is not needed here.

File: README.md [hunk 2]
```diff
@@ -18,3 +18,41 @@ error instead of silently succeeding (which `conda env` would do, thus
...
+
+To make matters worse, `conda env` often **silently succeeds**, which means you will just get random crashes because you have mixed incompatible and/or poorly specified (by other packages) dependencies.
+
+If you want to use conda envs `environment.yaml` then the only way to guarentee that you get the build from the channel you requested is to use hashed build strings. But if you are doing this, you might as well use explicit environment / lists of packages anyway.
+
+A 'known' [workaround](https://github.com/conda/conda/issues/6385#issuecomment-361787360) has been to just call `conda create` **not conda env** with all the packages in the environment yaml. **That's what this tool does**
+
+TL/DR: let's use the (hashed) build string to ensure conda gets our packages until channel pins in environment.yaml files get respected. From there, we should also try to use "locked" (aka "explicit") environments - although that is more of an upcoming conda feature than a reality in the conda 4.5 series.
```
Comment (sdvillal) [inline README.md]: I think this is not correct and I will contradict myself. While we use a conda env version that does not respect channel pins, we should simply use condaenv,

**Review by sdvillal — APPROVED** [review]

**Pull request merged**